tools/cldr-apps/src/main/java/org/unicode/cldr/web/Dashboard.java

```diff
@@ -170,53 +174,54 @@ public ReviewEntry(String code, String xpath) {
      */
     public ReviewOutput get(CLDRLocale locale, UserRegistry.User user, Level coverageLevel, String xpath) {
         final SurveyMain sm = CookieSession.sm;
-        String loc = locale.getBaseName();
         Organization usersOrg = Organization.fromString(user.voterOrg());
         STFactory sourceFactory = sm.getSTFactory();
-        final Factory baselineFactory = CookieSession.sm.getDiskFactory();
         VettingViewer<Organization> vv = new VettingViewer<>(sm.getSupplementalDataInfo(), sourceFactory,
             new STUsersChoice(sm));
-
         EnumSet<VettingViewer.Choice> choiceSet = VettingViewer.getChoiceSetForOrg(usersOrg);
+        DashboardArgs args = new DashboardArgs(choiceSet, locale, coverageLevel);
+        args.setUserAndOrganization(user.id, usersOrg);
+        setFiles(args, locale, sourceFactory);
+        if (xpath != null) {
+            args.setXpath(xpath);
+        }
+        return reallyGet(vv, args);
+    }
 
+    private void setFiles(DashboardArgs args, CLDRLocale locale, STFactory sourceFactory) {
         /*
          * sourceFile provides the current winning values, taking into account recent votes.
          * baselineFile provides the "baseline" (a.k.a. "trunk") values, i.e., the values that
          * are in the current XML in the cldr version control repository. The baseline values
          * are generally the last release values plus any changes that have been made by the
          * technical committee by committing directly to version control rather than voting.
          */
-        CLDRFile sourceFile = sourceFactory.make(loc);
-        CLDRFile baselineFile = baselineFactory.make(loc, true);
-        VettingViewer.DashboardArgs args = new VettingViewer.DashboardArgs(choiceSet, locale, coverageLevel);
-        args.setUserAndOrganization(user.id, usersOrg);
+        final String localeId = locale.getBaseName();
+        final CLDRFile sourceFile = sourceFactory.make(localeId);
+        final Factory baselineFactory = CookieSession.sm.getDiskFactory();
```

Comment:
for testability, there should be another overload of ReviewEntry() where the baselineFactory and stFactory are passed as parameters, instead of going through the static CookieSession.sm

Reply:
is there a ticket for that testability?